### PR TITLE
[WIP] AEA-531 Improvement/small refactoring to optimize tests

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes
depends on https://github.com/fetchai/agents-aea/pull/1085
small iprovements to run tests/test_aea.py faster
on my laptop speedup is about 20X
0.7 seconds  agains 17 seconds

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments
just an experimental tests update to learn more about agent, connections, messages workflow

### CODE REVIEW

File: aea/aea.py [hunk 1]
```diff
@@ -167,9 +167,13 @@ def react(self) -> None:
         counter = 0
         while not self.inbox.empty() and counter < self.max_reactions:
             counter += 1
-            envelope = self.inbox.get_nowait()  # type: Optional[Envelope]
-            if envelope is not None:
-                self._handle(envelope)
+            self._react_one()
+
+    def _react_one(self):
+        """ get and process one envelop from inbox """
```
Comment (DavidMinarsch) [inline aea/aea.py:173]: Please add return type ` -> None` and properly format the docstring:
```
"""
Get and process one envelop from inbox.

:return: None
"""
```

File: aea/aea.py [hunk 2]
```diff
@@ -167,9 +167,13 @@ def react(self) -> None:
         counter = 0
         while not self.inbox.empty() and counter < self.max_reactions:
             counter += 1
-            envelope = self.inbox.get_nowait()  # type: Optional[Envelope]
-            if envelope is not None:
-                self._handle(envelope)
+            self._react_one()
```
Comment (DavidMinarsch) [inline aea/aea.py:170]: Great!

File: aea/agent.py [hunk 3]
```diff
@@ -199,14 +199,20 @@ def start(self) -> None:
 
         :return: None
         """
+        self._start_setup()
+        self._run_main_loop()
+
+    def _start_setup(self) -> None:
+        """
```
Comment (DavidMinarsch) [inline aea/agent.py:206]: Please merge latest develop to get your changes from the other branch (now merged).

File: tests/test_aea.py [hunk 4]
```diff
@@ -170,16 +164,17 @@ def test_react():
         )
         builder.set_default_connection(PublicId.from_str("fetchai/local:0.1.0"))
         builder.add_skill(Path(CUR_PATH, "data", "dummy_skill"))
-        aea = builder.build(connection_ids=[PublicId.from_str("fetchai/local:0.1.0")])
+        agent = builder.build(connection_ids=[PublicId.from_str("fetchai/local:0.1.0")])
```
Comment (DavidMinarsch) [inline tests/test_aea.py:167]: As a rule aiding simpler reading, since we have both `AEA` and `Agent` it is best to refer to the class you're using, here `aea`

File: tests/common/utils.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    result = TimeItResult()
+    started_time = time.time()
+    try:
+        yield result
+    finally:
+        result.time_passed = time.time() - started_time
+
+
+class AgentTool:
+    def __init__(self, agent):
```
Comment (DavidMinarsch) [inline tests/common/utils.py:57]: Is this an agent or AEA? Please add function signature (types) and also rename class to `AEATool` if the `AEA` is used.

File: tests/common/utils.py [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    ) -> "AgentTool":
+        """ wait till agent's outbox consumed completely """
+        start_time = time.time()
+        while not self.agent.outbox.empty():
+            time.sleep(sleep)
+            if time.time() - start_time > timeout:
+                raise Exception("timeout")
+        return self
+
+    def wait_inbox(self, sleep=DEFAULT_SLEEP, timeout=DEFAULT_TIMEOUT) -> "AgentTool":
```
Comment (DavidMinarsch) [inline tests/common/utils.py:81]: please always declare types, i.e. `sleep: str = DEFAULT_SLEEP`

File: tests/common/utils.py [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+class AgentTool:
+    def __init__(self, agent):
+        self.agent = agent
+
+    def setup(self) -> "AgentTool":
+        self.agent._start_setup()
+        return self
+
+    def spin(self) -> "AgentTool":
```
Comment (DavidMinarsch) [inline tests/common/utils.py:64]: maybe call it `spin_main_loop` so it is clearer?

File: tests/test_aea.py [hunk 8]
```diff
@@ -192,26 +187,30 @@ def test_react():
...
+            """ inbox twice cause first message is invalid. generates error message and it accepted """
+            tool.wait_inbox().react_one()
+            tool.wait_inbox().react_one()
             assert len(dummy_handler.handled_messages) == 1
 
             #   DECODING ERROR
             envelope = Envelope(
-                to=aea.identity.address,
-                sender=aea.identity.address,
+                to=agent.identity.address,
```
Comment (DavidMinarsch) [inline tests/test_aea.py:204]: see above

File: tests/test_aeabuilder.py [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+    """
+    agent_name = "MyAgent"
+    private_key_path = os.path.join(CUR_PATH, "data", "fet_private_key.txt")
+    builder = AEABuilder()
+    builder.set_name(agent_name)
+    builder.add_private_key(FETCHAI, private_key_path)
+    builder.DEFAULT_AGENT_LOOP_TIMEOUT = 0.05
+
+    """ Default timeout == 0.05 """
+    agent = builder.build()
```
Comment (DavidMinarsch) [inline tests/test_aeabuilder.py:43]: See above, better to call it `aea`

File: tox.ini [hunk 10]
```diff
@@ -25,7 +25,7 @@ commands =
     pip install git+https://github.com/pytoolz/cytoolz.git#egg=cytoolz==0.10.1.dev0
     pip install -e .[all]
     pip install -i https://test.pypi.org/simple/ fetch-p2p-api==0.0.2
-    pytest --doctest-modules aea packages/fetchai/protocols packages/fetchai/connections tests/ --cov-report=html --cov-report=xml --cov-report=term --cov=aea --cov=packages/fetchai/protocols --cov=packages/fetchai/connections {posargs}
+    pytest --doctest-modules tests/test_aea.py --cov-report=html --cov-report=xml --cov-report=term --cov=aea --cov=packages/fetchai/protocols --cov=packages/fetchai/connections {posargs}
```
Comment (DavidMinarsch) [inline tox.ini:28]: please revert this change

Comment (solarw) [reply tox.ini:28]: sorry, this done by mistake

**Comment by solarw** [comment]
> LGTM, I made a few suggestions to aid consistency

updated

File: .github/workflows/workflow.yml [hunk 11]
```diff
@@ -8,51 +8,7 @@ on:
   pull_request:
 
 jobs:
-  common_checks:
```
Comment (DavidMinarsch) [inline .github/workflows/workflow.yml:11]: why are you reverting this change?

Comment (solarw) [reply .github/workflows/workflow.yml]: I'm curious about this change, cause it supposed to be in develop, and this file was synced with develop branch, probably my develop branch was not synced at the moment
